### PR TITLE
doc: update documentation on release

### PR DESCRIPTION
**Pull request by ktaborowski**
[KRKNWK-17045]

### CODE REVIEW

File: doc/sidewalk_release_notes.rst [hunk 1]
```diff
@@ -5,6 +5,16 @@ Release notes
 
 See the release notes for the information about specific Sidewalk support releases for nRF Connect SDK.
 
+[v1.14.4] - 06/XX/2023
```
Comment (ktaborowski) [inline doc/sidewalk_release_notes.rst:8]: 'XX' will be changed on release date when knwon

File: doc/sidewalk_release_notes.rst [hunk 2]
```diff
@@ -5,6 +5,15 @@ Release notes
...
+----------------------
+
+* Rleased Sidewalk based on the nRF Connect SDK v2.4.0.
```
Comment (annwoj) [inline doc/sidewalk_release_notes.rst:11]:
```suggestion
* Released Sidewalk based on the nRF Connect SDK v2.4.0.
```

File: doc/sidewalk_release_notes.rst [hunk 3]
```diff
@@ -5,6 +5,15 @@ Release notes
...
+* Moved Semtech pins configuration to DTS.
```
Comment (annwoj) [inline doc/sidewalk_release_notes.rst:12]:
```suggestion
* Moved the Semtech pins configuration to DTS.
```

File: doc/sidewalk_release_notes.rst [hunk 4]
```diff
@@ -5,6 +5,15 @@ Release notes
...
+* Changed DFU mode bahvaiour. Device restes before enter the mode.
```
Comment (annwoj) [inline doc/sidewalk_release_notes.rst:13]:
```suggestion
* Changed the DFU mode behavior. 
  The device now resets before entering the DFU mode.
```

File: doc/sidewalk_release_notes.rst [hunk 5]
```diff
@@ -5,6 +5,15 @@ Release notes
...
+* Notified DFU state with LEDs.
```
Comment (annwoj) [inline doc/sidewalk_release_notes.rst:14]:
```suggestion
* Added notifying through LEDs whenever the device enters the DFU mode.
```

Rephrased a bit, please let me know if the meaning is affected

File: doc/sidewalk_release_notes.rst [hunk 6]
```diff
@@ -5,6 +5,15 @@ Release notes
...
+* Automated repository maintenace with GitHub Actions.
```
Comment (annwoj) [inline doc/sidewalk_release_notes.rst:15]:
```suggestion
* Automated the repository maintenance with GitHub Actions.
```

**Review by annwoj — CHANGES_REQUESTED** [review]

**Review by annwoj — APPROVED** [review]

File: doc/sidewalk_release_notes.rst [hunk 7]
```diff
@@ -5,6 +5,16 @@ Release notes
...
+[v1.14.4] - 06/05/2023
+----------------------
+
+* Released Sidewalk based on the nRF Connect SDK v2.4.0.
+* Moved the Semtech pins configuration to DTS.
+* Changed the DFU mode behavior.
+  The device now resets before entering the DFU mode.
+* Added notifying through LEDs whenever the device enters the DFU mode.
+* Automated the repository maintenance with GitHub Actions.
+
```
Comment (totyz) [inline doc/sidewalk_release_notes.rst:17]:
```suggestion
* show link status for every transport in serial log
* Add [aws iot core tools](https://github.com/aws-samples/aws-iot-core-for-amazon-sidewalk-sample-app) to the west.yml manifest file
```

**Review by totyz — CHANGES_REQUESTED** [review]

File: doc/sidewalk_release_notes.rst (hunk 7 above)
Comment (ktaborowski) [reply doc/sidewalk_release_notes.rst:19]: Done

File: doc/sidewalk_release_notes.rst [hunk 8]
```diff
@@ -5,6 +5,18 @@ Release notes
...
+----------------------
+
+* Released Sidewalk based on the nRF Connect SDK v2.4.0.
+* Moved the Semtech pins configuration to DTS.
+* Changed the DFU mode behavior.
+  The device now resets before entering the DFU mode.
+* Added notifying through LEDs whenever the device enters the DFU mode.
+* Automated the repository maintenance with GitHub Actions.
+* Added link status for every transport in serial log.
+* Added aws iot core tools repostiory to the west manifest file.
```
[the remark this answers is not in the document]
Comment (ktaborowski) [reply doc/sidewalk_release_notes.rst]: Link added, thanks for suggestions

**Review by totyz — APPROVED** [review]

**Pull request merged**